File: core/src/main/java/com/cc/core/xposed/BaseXposedHook.java
package com.cc.core.xposed;

import de.robv.android.xposed.XposedHelpers;

public abstract class BaseXposedHook implements XposedHook {
    protected void hookMethod(String className, ClassLoader classLoader, String methodName, Object... params) {
        XposedHelpers.findAndHookMethod(className, classLoader, methodName, params);
    }

    protected Class<?> findClass(String className, ClassLoader classLoader) {
        return XposedHelpers.findClass(className, classLoader);
    }
}

File: core/src/main/java/com/cc/core/xposed/BaseXposedHook.kt
package com.cc.core.xposed

import de.robv.android.xposed.XC_MethodHook
import de.robv.android.xposed.XposedHelpers

abstract class BaseXposedHook : XposedHook {
    fun hookMethod(className: String, classLoader: ClassLoader, methodName: String, vararg params: Any) : XC_MethodHook.Unhook  {
        return XposedHelpers.findAndHookMethod(className, classLoader, methodName, *params)
    }

    fun findClass(className: String, classLoader: ClassLoader): Class<*> {
        return XposedHelpers.findClass(className, classLoader)
    }
}